src/components/datagrid/utils/col_widths.ts

```diff
@@ -56,21 +53,12 @@ export const useDefaultColumnWidth = (
     return Math.max(widthToFill / unsizedColumnCount, DEFAULT_COLUMN_WIDTH);
   }, [gridWidth, columns, leadingControlColumns, trailingControlColumns]);
 
-  const [defaultColumnWidth, setDefaultColumnWidth] = useState<number | null>(
-    computeDefaultWidth
-  );
-
-  useEffect(() => {
-    const columnWidth = computeDefaultWidth();
-    setDefaultColumnWidth(columnWidth);
-  }, [computeDefaultWidth]);
-
   return defaultColumnWidth;
 };
 
 export const doesColumnHaveAnInitialWidth = (
   column: EuiDataGridColumn
-): column is EuiDataGridColumn & { initialWidth: number } => {
+): boolean => {
   return column.hasOwnProperty('initialWidth');
```

Comment:
Unrelated to your changes other than you're already at this function, this should also ensure the initialWidth value isn't `undefined`

```suggestion
  return column.hasOwnProperty('initialWidth') && column.initialWidth != undefined;
```

Reply:
Done: https://github.com/elastic/eui/pull/5689/commits/b91cb335f81ffb55f517e597ab9ee584b2b40c71

FYI that I had to use `!= null` instead of `!= undefined`, eslint complained otherwise (and I figured it's nice to capture both null and undefined)